src/python/platforms/fuchsia/undercoat.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Helper functions for managing and interacting with Fuchsia devices
+via undercoat."""
+
+import os
+import sys
+import tempfile
+
+from metrics import logs
+from system import environment
+from system import new_process
+from system import shell
+
+# In order to properly clean up any stale instances, we need to keep track of
+# handles in a way that persists between runs
+UNDERCOAT_HANDLE_DIR = os.path.join(tempfile.gettempdir(), 'undercoat-handles')
```

Comment:
put this into a function rather than a global, and use environment.get_value('BOT_TMPDIR') instead.

i.e.

```python
def get_handle_dir():
  return os.path.join(...)
```

Reply:
Ah, nice! The only thing `persistent_cache` doesn't allow that I was doing before is enumeration, but I think it's unnecessary if I don't need to worry about multiple processes acting on the same cache key simultaneously.  I assume this is a reasonable assumption to make?